src/components/common/PaginatedTxns/index.tsx

```diff
@@ -9,93 +11,78 @@ import InfiniteScroll from '../InfiniteScroll'
 import SkeletonTxList from './SkeletonTxList'
 import BatchExecuteButton from '@/components/transactions/BatchExecuteButton'
 import TxFilterButton from '@/components/transactions/TxFilterButton'
-import { TxFilter, useTxFilter } from '@/utils/tx-history-filter'
-import { isTransactionListItem } from '@/utils/transaction-guards'
-import type { TransactionListPage } from '@gnosis.pm/safe-react-gateway-sdk'
+import { type TxFilter, useTxFilter } from '@/utils/tx-history-filter'
 import { BatchExecuteHoverProvider } from '@/components/transactions/BatchExecuteButton/BatchExecuteHoverProvider'
+import { isTransactionListItem } from '@/utils/transaction-guards'
 
 const NoQueuedTxns = () => (
   <Box mt="5vh">
     <PagePlaceholder imageUrl="/images/no-transactions.svg" text="Queued transactions will appear here" />
   </Box>
 )
 
-const getResultCount = (filter: TxFilter, page: TransactionListPage) => {
+const getFilterResultCount = (filter: TxFilter, page: TransactionListPage) => {
   const count = page.results.filter(isTransactionListItem).length
 
   return `${page.next ? '> ' : ''}${count} ${filter.type} transactions found`.toLowerCase()
 }
 
-const TxPage = ({
-  pageUrl,
-  useTxns,
-  onNextPage,
-  isFirstPage,
-}: {
-  pageUrl: string
-  useTxns: typeof useTxHistory | typeof useTxQueue
-  onNextPage?: (pageUrl?: string) => void
-  isFirstPage: boolean
-}): ReactElement => {
-  const { page, error } = useTxns(pageUrl)
+const PaginatedTxns = ({ useTxns }: { useTxns: typeof useTxHistory | typeof useTxQueue }): ReactElement => {
+  const [pages, setPages] = useState<Array<{ pageUrl?: string; data: TransactionListPage }>>([])
+  const [pageUrl, setPageUrl] = useState<string>()
   const [filter] = useTxFilter()
-
   const isQueue = useTxns === useTxQueue
+  const allItems = useMemo(() => flatten(pages.map((page) => page.data.results)), [pages])
```

Comment:
I suggest we use [`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) instead of `lodash`.

Reply:
Oh wow, didn’t know we had this in JS, thanks!